code/modules/projectiles/projectile/experimental.dm

```diff
@@ -0,0 +1,681 @@
+		if(Y>0)
+			if(X>Y)
+				pos_from = SIDE_WEST
+				pos_to = SIDE_NORTH
+				X_spawn = 1
+			else if(X<Y)
+				pos_from = SIDE_SOUTH
+				pos_to = SIDE_EAST
+				Y_spawn = 1
+			else
+				if(prob(50))
+					pos_from = SIDE_WEST
+					pos_to = SIDE_NORTH
+					X_spawn = 1
+				else
+					pos_from = SIDE_SOUTH
+					pos_to = SIDE_EAST
+					Y_spawn = 1
+		else if(Y<0)
+			if(X>(Y*-1))
+				pos_from = SIDE_WEST
+				pos_to = SIDE_SOUTH
+				X_spawn = 1
+			else if(X<(Y*-1))
+				pos_from = SIDE_NORTH
+				pos_to = SIDE_EAST
+				Y_spawn = -1
+			else
+				if(prob(50))
+					pos_from = SIDE_WEST
+					pos_to = SIDE_SOUTH
+					X_spawn = 1
+				else
+					pos_from = SIDE_NORTH
+					pos_to = SIDE_EAST
+					Y_spawn = -1
+		else if(Y==0)
+			pos_from = SIDE_WEST
+			X_spawn = 1
+			if(prob(50))
+				pos_to = SIDE_NORTH
+			else
+				pos_to = SIDE_SOUTH
+	else if(X<0)
+		if(Y>0)
+			if((X*-1)>Y)
+				pos_from = SIDE_EAST
+				pos_to = SIDE_NORTH
+				X_spawn = -1
+			else if((X*-1)<Y)
+				pos_from = SIDE_SOUTH
+				pos_to = SIDE_WEST
+				Y_spawn = 1
+			else
+				if(prob(50))
+					pos_from = SIDE_EAST
+					pos_to = SIDE_NORTH
+					X_spawn = -1
+				else
+					pos_from = SIDE_SOUTH
+					pos_to = SIDE_WEST
+					Y_spawn = 1
+		else if(Y<0)
+			if((X*-1)>(Y*-1))
+				pos_from = SIDE_EAST
+				pos_to = SIDE_SOUTH
+				X_spawn = -1
+			else if((X*-1)<(Y*-1))
+				pos_from = SIDE_NORTH
+				pos_to = SIDE_WEST
+				Y_spawn = -1
+			else
+				if(prob(50))
+					pos_from = SIDE_EAST
+					pos_to = SIDE_SOUTH
+					X_spawn = -1
+				else
+					pos_from = SIDE_NORTH
+					pos_to = SIDE_WEST
+					Y_spawn = -1
+		else if(Y==0)
+			pos_from = SIDE_EAST
+			X_spawn = -1
+			if(prob(50))
+				pos_to = SIDE_NORTH
+			else
+				pos_to = SIDE_SOUTH
+	else if(X==0)
+		if(Y>0)
+			Y_spawn = 1
+			pos_from = SIDE_SOUTH
+			if(prob(50))
+				pos_to = SIDE_EAST
+			else
+				pos_to = SIDE_WEST
+		else if(Y<0)
+			Y_spawn = -1
+			pos_from = SIDE_NORTH
+			if(prob(50))
+				pos_to = SIDE_EAST
+			else
+				pos_to = SIDE_WEST
+	else
+		OnDeath()
+		loc = null
+		returnToPool(src)
+		return
+
+	var/turf/newspawn = locate(T1.x + X_spawn, T1.y + Y_spawn, z)
+	src.loc = newspawn
+
+	update_icon()
+
+/obj/item/projectile/ricochet/update_icon()//8 possible combinations
+	switch(pos_to)
+		if(SIDE_NORTH)
+			if(pos_from == SIDE_WEST)
+				dir = NORTHWEST
+			else
+				dir = EAST
+		if(SIDE_SOUTH)
+			if(pos_from == SIDE_WEST)
+				dir = WEST
+			else
+				dir = SOUTHEAST
+		if(SIDE_EAST)
+			if(pos_from == SIDE_NORTH)
+				dir = NORTHEAST
+			else
+				dir = SOUTH
+		if(SIDE_WEST)
+			if(pos_from == SIDE_NORTH)
+				dir = NORTH
+			else
+				dir = SOUTHWEST
+
+/obj/item/projectile/ricochet/Bump(atom/A as mob|obj|turf|area)
+	if(bumped)	return 0
+	bumped = 1
+
+	if(A)
+		if(istype(A,/turf/) || (istype(A,/obj/machinery/door/) && A.opacity))
+			var/obj/structure/ricochet_bump/bump = new(loc)
+			playsound(get_turf(src), 'sound/items/metal_impact.ogg', 50, 1)
+			switch(pos_to)
```

Comment:
right. I can do that. now that I've replaced the strings with defines. I. need some coffee.